docker/client.py

```diff
@@ -140,9 +140,14 @@ def build(self, path=None, tag=None, quiet=False, fileobj=None,
             if self._auth_configs:
                 if headers is None:
                     headers = {}
-                headers['X-Registry-Config'] = auth.encode_full_header(
-                    self._auth_configs
-                )
+                if utils.compare_version('1.19', self._version) >= 0:
```

Comment:
I tested my setup using docker version 1.6.2, docker-compose 1.3.3, along with a patched version of docker-py, and I can verify it did work correctly with the change in this pull request. This looks like the cleanest way for docker-py to implement this change. It looks like I'm going to have to wait for docker-compose to change the supported API to 1.19.